### ISSUE TITLE
object member incorrectly typed as unknown

### ISSUE THREAD

**Issue by chdsbd** [microsoft/pyright#935]
**Describe the bug**
An object member is incorrectly typed as unknown when it should be typed as `def ()`.

**To Reproduce**

With django installed open the following code and see that `serialize` is incorrectly marked as unknown.
```python
from django.db import models

class Manager(models.Manager):
    def serialize(self) -> None:
        pass

class Model(models.Model):
    objects = Manager()

Model.objects.serialize() # see that serialize is `Unknown`.
```

error
```
Cannot access member "serialize" for type "BaseManager[Any]"
  Member "serialize" is unknownPylance (reportGeneralTypeIssues)
```

**Expected behavior**
`Model.objects.serialize` should be typed as `def ()` instead of `Unknown`, like mypy types it.

**Screenshots or Code**
<img width="536" alt="image" src="https://user-images.githubusercontent.com/1929960/89941816-944a9480-dbe9-11ea-9f97-e9348b72bc10.png">

**VS Code extension or command-line**
I'm using Pylance v2020.8.0.

**Additional context**

**Comment by erictraut** [comment]
Pyright is doing the right thing here. The type stub for django provides no definition for a `serialize` attribute in the `BaseManager` class or its parent classes, so the type checker is correctly flagging this. That means either `serialize` is not part of the public interface contract or (more likely) the django type stub is incomplete. If you think it's the latter, please file a bug to that effect in the "pylance-release" repo. Thanks!

**Comment by jakebailey** [comment]
But both types are declared in this file; the `Model` class is assigning `objects` the member with the `Manager` type declared above.

Is it expected that this type of assignment (at the top level, not as a part of a function or similar) does not change the type of the `objects` member?

**Comment by jakebailey** [comment]
Normally I'd think that this is a "well the base class said it was one thing", but I can also see the point of view that the `Manager` class can replace the `BaseManager` member without breaking any rules, and then uses of `Model` will be (at least) that new type instead.

**Comment by erictraut** [comment]
Ah, I missed that point. The issue is that the base class declared that the type of `objects` is `BaseManager[Any]`. Your subclass assigned the value, but it didn't provide any type annotation, so Pyright uses the declared type. If you want to redeclare the type to a narrower type within your subclass, you can do so:

```
class Model(models.Model):
    objects: Manager = Manager()
```
```

**Comment by chdsbd** [comment]
Thanks for the suggestion @erictraut, adding the type hint fixed the issue!
